src/jestUtils.ts

```diff
@@ -0,0 +1,491 @@
+      const shouldTransitionToNextState = (nextEvent: Event) =>
+        noEventsLeft(nextEvent) ||
+        hasState(State.END)(nextEvent) ||
+        hasState(State.FAILED)(nextEvent) ||
+        hasState(State.CANCELLED)(nextEvent);
+
+      fillEventsForCurrentState({
+        shouldConsumeEvent,
+        shouldTransitionToNextState,
+      });
+    } else if (nextRequiredState === State.END) {
+      fillEventsForCurrentState({});
+    }
+    hasAllStates = currentStateIdx === REQUIRED_EVENTS.length;
+
+    invariant(
+      iterations++ <= 500,
+      'exceeded max number of iterations, please report a bug in RNGH repository with your test case'
+    );
+  } while (!hasAllStates);
+
+  return transformedEvents;
+}
+
+type EventEmitter = (
+  eventName: string,
+  args: { nativeEvent: GestureHandlerTestEvent }
+) => void;
+interface HandlerData {
+  emitEvent: EventEmitter;
+  handlerType: HandlerNames;
+  handlerTag: number;
+}
+function getHandlerData(
+  componentOrGesture: ReactTestInstance | GestureType
+): HandlerData {
+  if (isGesture(componentOrGesture)) {
+    const gesture = componentOrGesture;
+    return {
+      emitEvent: (eventName, args) => {
+        DeviceEventEmitter.emit(eventName, args.nativeEvent);
+      },
+      handlerType: gesture.handlerName as HandlerNames,
+      handlerTag: gesture.handlerTag,
+    };
+  }
+  const gestureHandlerComponent = componentOrGesture;
+  return {
+    emitEvent: (eventName, args) => {
+      fireEvent(gestureHandlerComponent, eventName, args);
+    },
+    handlerType: gestureHandlerComponent.props.handlerType as HandlerNames,
+    handlerTag: gestureHandlerComponent.props.handlerTag as number,
+  };
+}
+type AllGestures =
+  | TapGesture
+  | PanGesture
+  | LongPressGesture
+  | RotationGesture
+  | PinchGesture
+  | FlingGesture
+  | ForceTouchGesture
+  | NativeGesture;
+
+type AllHandlers =
+  | TapGestureHandler
+  | PanGestureHandler
+  | LongPressGestureHandler
+  | RotationGestureHandler
+  | PinchGestureHandler
+  | FlingGestureHandler
+  | ForceTouchGestureHandler
+  | NativeViewGestureHandler;
+
+// prettier-ignore
+type ClassComponentConstructor<P> = new (props: P) => React.Component<P, any, any>;
+
+type ExtractPayloadFromProps<T> = T extends BaseGestureHandlerProps<
+  infer TPayload
+>
+  ? TPayload
+  : never;
+
+type ExtractConfig<T> = T extends BaseGesture<infer TGesturePayload>
+  ? TGesturePayload
+  : T extends ClassComponentConstructor<infer THandlerProps>
+  ? ExtractPayloadFromProps<THandlerProps>
+  : Record<string, unknown>;
+
+export function fireGestureHandler<THandler extends AllGestures | AllHandlers>(
+  componentOrGesture: ReactTestInstance | GestureType,
+  eventList: Partial<GestureHandlerTestEvent<ExtractConfig<THandler>>>[] = []
+): void {
+  const { emitEvent, handlerType, handlerTag } = getHandlerData(
+    componentOrGesture
+  );
+
+  let _ = fillMissingStatesTransitions(
+    eventList,
+    isDiscreteHandler(handlerType)
+  );
```

Comment:
Why `_` as a name?

Reply:
In JS we don't have a pipe operator and this is basically a pipeline. Adding meaningful names would only obfuscate this code since we don't need intermediate results. Another approach is to modify the prototype (https://github.com/software-mansion/react-native-gesture-handler/pull/1844/commits/f28cb394c3156dd6a3341cae456f9a7ea54d2165) and chain it but it's rarely seen in JS codebases so I reverted to this one.